navigator/backend/src/main/scala/com/digitalasset/navigator/store/platform/PlatformSubscriber.scala

```diff
@@ -80,20 +81,24 @@ class PlatformSubscriber(
     } yield {
       Started(tracker)
     }
-
-    started onComplete {
-      case Success(value) =>
-        log.debug("Started actor for '{}'", party.name)
-        self ! value
-      case Failure(error) =>
-        // Failed to start up, giving up
-        log.error(
-          "Failed to start actor for party '{}': {}. Please fix any issues and restart this application.",
-          party.name,
-          error,
-        )
-        context.become(failed(error))
-    }
+    import scalaz.std.scalaFuture._
+    import scalaz.syntax.monadError._
+    import scalaz.{-\/, \/-}
+    val it =
+      started.attempt.map {
+        case \/-(value) =>
+          log.debug("Started actor for '{}'", party.name)
+          self ! value
+        case -\/(error) =>
+          // Failed to start up, giving up
+          log.error(
+            "Failed to start actor for party '{}': {}. Please fix any issues and restart this application.",
+            party.name,
+            error,
+          )
+          context.become(failed(error))
+      }
+    scala.concurrent.Await.ready(it, scala.concurrent.duration.Duration.fromNanos(300000000000L))
```

Comment:
```suggestion
    import scalaz.std.scalaFuture._
    import scalaz.syntax.monadError._
    import scalaz.{-\/, \/-}
    import scala.concurrent.Await
    import scala.concurrent.duration.DurationInt
    val actorStart =
      started.attempt.map {
        case \/-(value) =>
          log.debug("Started actor for '{}'", party.name)
          self ! value
        case -\/(error) =>
          // Failed to start up, giving up
          log.error(
            "Failed to start actor for party '{}': {}. Please fix any issues and restart this application.",
            party.name,
            error,
          )
          context.become(failed(error))
      }
    Await.ready(actorStart, 300.seconds)
```
1. let's avoid names like `it`
1. check that this compiles, I wrote
2. I left the value to 300 seconds by removing nine trailing zeros, I'm not sure whether you actually wanted 300 or wanted 30 and added an extra 0

Reply:
This touches on an important, more general principle: only change what really needs to be changed, and make sure your change follows the local convention. As the old code uses standard Scala `Future`/`Try`, please keep using that. If there's a compelling reason to start using scalaz, please explain why.

Reply:
I usually make use of scalaz because we use it all over the repository (especially within the json api) plus it makes code that works with futures easier to handle & read thanks to some simple but helpful methods like `attempt`. However will adjust that code to not use scalaz anymore if you think this is misused here.

Reply:
adjusted